### PR TITLE
Add support for overlay function.

### LINKED ISSUE #225: Document the overlay function
### Related code PR

https://github.com/singularity-data/risingwave/pull/3671

### Which part(s) of the docs might be affected or should be updated? And how?

SQL reference - > Functions and operators -> String -> String functions

### Reference

https://www.postgresql.org/docs/9.1/functions-string.html

### PR DESCRIPTION
**Pull request by bernscode**
Add support for overlay function.

## Info
- **Description**: 
Add support for overlay function.

- **Related code PR**: 
https://github.com/singularity-data/risingwave/pull/3671

- **Related doc issue**: 
Resolves https://github.com/singularity-data/risingwave-docs/issues/225

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [x] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview and the updated parts look all good. <details><summary>How?</summary>image</details>
  - [ ] (For version-specific PR) I have confirmed that this PR is for a released version. If the software version is not released, put it on hold.

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Comment by ALeitert** [comment]
I think we should write `[ FOR int ]` to make it more clear that this part is optional.

Could we rewrite the description to refer to variables (similar to the description of Repeat or Substr) and to state how it works if the last part is omitted? I was thinking about something like this:
> Replaces a substring in *string*, starting at *start* and with length *len*, with *substr*. If *len* is omitted, its value is the length of *substr*.

**Comment by bernscode** [comment]
> 

Great point, @ALeitert! I've made the recommended changes. Thanks!

**Comment by ALeitert** [comment]
LGTM!

**Review by ALeitert — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]
LGTM

File: docs/sql/functions-operators/sql-function-string.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ title: String functions and operators
...
 |CONCAT_WS (sep varchar, val1 any [, val2 any [, ...] ]) → varchar | Concatenates the arguments with a separator. The first argument is used as the separator, and should not be NULL. Other NULL arguments are ignored. | `CONCAT_WS(',', 'Abcde', 2, NULL, 22)` → `Abcde,2,22` |
 |SPLIT_PART (string varchar, delimiter varchar, n int) → varchar | Splits *string* at occurrences of *delimiter* and returns the *n*'th field (counting from one), or when *n* is negative, returns the \|*n*\|'th-from-last field. When n is zero, returns an 'InvalidParameterValue' error. When the input *delimiter* is an empty string, returns the input *string* if querying the first or last field. Otherwise, returns an empty string. | `SPLIT_PART('abc~@~def~@~ghi', '~@~', 2)` → `def` |
+|OVERLAY(string PLACING substr string FROM start int [ FOR len int ]) → string | Replaces a substring in string with substr, starting at start and with length len. If len is omitted, its value is the length of substr. | `OVERLAY('yabadoo' PLACING 'daba' FROM 5 FOR 0)` → `yabadabadoo` <br /> `OVERLAY('abcdef' PLACING '45' FROM 4)` → `abc45f`|
```
Comment (CharlieSYH) [inline docs/sql/functions-operators/sql-function-string.md]: My personal preference is to italicize the parameter name. For example, `Replaces a substring in *string* with *substr*, starting at *start* and with length *len*.` I believe it improves readability.

Comment (CharlieSYH) [reply docs/sql/functions-operators/sql-function-string.md]: Yabadabadoo, huh. Interesting. https://www.youtube.com/watch?v=pjfWwHAILQk

Comment (bernscode) [reply docs/sql/functions-operators/sql-function-string.md]: @CharlieSYH Good point! I agree! I'll make the recommended changes, commit, and merge. Thanks!

**Pull request merged**